### DISCUSSION TITLE
could anyone explain what shapetracker does

### DISCUSSION THREAD

**Discussion by ningyuwhut** [tinygrad/tinygrad#394] [Q&A]
I'm been reading the code since several days ago and it seems a little hard for me. I'm wondering are there some materails to help understand it, I searched the web and find barelly no articles or blogs about this repo except the youtube videos, but the videos are so long and it's hard to find which part is related to my questions. For example, I can't understand what the shapetracker does. can someone explain it. Any replies will be appreciated.

**Comment by DKormann** [comment]
as far as I understand shapetracker contains shape and strides of a Tensor.
This information is needed in lazy.movement_op.
to understand strides take a look at [this](https://en.wikipedia.org/wiki/Stride_of_an_array) Wiki article
